src/stage1/image_retrieval.rs

```diff
@@ -166,6 +145,81 @@ fn determine_version(ver_str: &str, versions: &Versions) -> Result<Version> {
     }
 }
 
+#[cfg(test)]
+mod determine_version {
```

Comment:
From [here](https://doc.rust-lang.org/book/ch11-03-test-organization.html#unit-tests):

> The convention is to create a module named tests in each file to contain the test functions [...]

```suggestion
mod tests {
```

(Again, love that you added tests! :slightly_smiling_face:)

Reply:
Yeah, i was not too sure! 
I liked how it logs `test stage1::image_retrieval::determine_version::returns_latest_version_by_default ... ok` with the module name matching the method name compared to `test stage1::utils::tests::test_read_buffer ... ok`

but let's stick to conventions :) thanks